bouncerscript/task.py

```diff
@@ -59,3 +63,12 @@ def validate_task_schema(context):
     action = get_task_action(context.task, context.config)
     schema_key = "schema_files.{}".format(action)
     client.validate_task_schema(context, schema_key=schema_key)
+
+
+def preflight_check(context):
+    """Make sure we don't do any cross-product/channel alias update"""
+    aliases = context.task["payload"]["aliases_entries"]
+    allowed_regexes = context.config['aliases_regexes']
+
+    return all([matches(product_name, allowed_regexes[alias])
```

Comment:
We may die with a `KeyError` here if the alias isn't in `allowed_regexes`. We may want to check for that and raise a `TaskVerificationError`.